astropy/stats/histogram.py

```diff
@@ -0,0 +1,332 @@
+"""
+Methods for selecting the bin width of histograms
+
+Ported from the astroML project: http://astroML.org/
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from ..extern import six
+
+import numpy as np
+from . import bayesian_blocks
+
+__all__ = ['histogram', 'scotts_bin_width', 'freedman_bin_width',
+           'knuth_bin_width']
+
+
+def histogram(a, bins=10, range=None, weights=None, **kwargs):
+    """Enhanced histogram function
+
+    This is a histogram function that enables the use of more sophisticated
+    algorithms for determining bins.  Aside from the ``bins`` argument allowing
+    a string specified how bins are computed, the parameters are the same
+    as ``numpy.histogram()``.
+
+    Parameters
+    ----------
+    a : array_like
+        array of data to be histogrammed
+
+    bins : int or list or str (optional)
```

Comment:
Yes between each of the items so that:

    bins : int or list or str (optional)
        If bins is a string, then it must be one of:

        * 'adaptive' : use bayesian blocks for dynamic bin widths

        * 'blocks' : same as 'adaptive'

        * 'knuth' : use Knuth's rule to determine bins

        * 'scotts' : use Scott's rule to determine bins

        * 'freedman' : use the Freedman-Diaconis rule to determine bins


Otherwise the html sphinx page ends up compress it all down into a single block of text with no itemized list.